### PR TITLE
disable padding=same for experimental hook

### PR DESCRIPTION
**Pull request by ffuuugor**
The tests are failing because our experimental `convolution as a backward` GSM can't handle `padding="same"`

Since this grad sampler is not used by default and mostly exists for aesthetic reasons, I think disabling the test is the best course of action.

### CODE REVIEW

File: opacus/tests/grad_samples/conv2d_test.py [hunk 1]
```diff
@@ -85,20 +85,22 @@ def test_conv2d(
...
-            x,
-            conv,
-            batch_first=True,
-            atol=10e-5,
-            rtol=10e-4,
-            ew_compatible=is_ew_compatible,
-        )
-        GradSampleModule.GRAD_SAMPLERS[nn.Conv2d] = conv2d_gsm
+
+        if padding != "same":
```
Comment (karthikprasad) [inline opacus/tests/grad_samples/conv2d_test.py:89]: Can you add a comment here explaining why this check is needed? Essentially copy-pasta the PR description/summary here (and may be also in the actual grad_sampler file)

**Review by karthikprasad — APPROVED** [review]
Left one non-blocking comment. Thanks for the quick fix :)
